armbian/base/scripts/prometheus-base.py

```diff
@@ -80,6 +81,27 @@ def getSystemdStatus(unit):
         print(unit, e.returncode, e.output)
         return e.returncode
 
+def getInternetConnectivity():
+    torEnabled = int(r.get('tor:base:enabled').decode("utf-8"))
+
+    try:
+        if torEnabled == 1:
+            print("Tor ok")
+            subprocess.check_output(["curl", "--socks5-hostname", "localhost:9050", "1.1.1.1"], shell=False, timeout=5, stderr=subprocess.STDOUT)
+        else:
+            print("Tor not ok")
+            subprocess.check_output(["ping", "-c", "1", "1.1.1.1"], shell=False, timeout=5, stderr=subprocess.STDOUT)
+
+        return 0
+
+    except subprocess.TimeoutExpired as e:
+        print("getInternetConnectivity(): subprocess.TimeoutExpired; torEnabled", torEnabled)
+        return 1
+
+    except subprocess.CalledProcessError as e:
+        print("getInternetConnectivity(): subprocess.CalledProcessError (", e.returncode, "); torEnabled", torEnabled, e.output)
+        return e.returncode
```

Comment:
> provides the new Prometheus metric 'base_internet_connectivity' that is 0 when OK, or an error code when NOT OK

I think that's a bit unintuitive to use 0 as OK and everything else as NOK. Either rename `base_internet_connectivity` to `base_no_internet_connectivity` or even better: return 1 if internet and 0 if no internet connectivity. Does curl or ping even return detailed info in the exit code?

Reply:
I agree that it's not very intuitive and will change it. 

I followed the logic of `getSystemdStatus`, which returns `0` if service is running. Maybe change this as well? Not sure, where that code is used elsewhere, in Supervisor or Middleware.

Reply:
Commit https://github.com/digitalbitbox/bitbox-base/pull/301/commits/f64a68b121df419d17af090a5fba440940a39bfd changes logged values for both "network connectivity" and "systemd units" and updates the Grafana dashboard to reflect these changes.

Reply:
Post-merge nACK from my side.
I really disagree that it's `unintuitive to use 0 as OK and everything else as NOK`, afaik that is the default convention in most systems. Definitely in all UNIX/POSIX systems and in C running on these systems. (https://en.wikipedia.org/wiki/Exit_status)
I'd say it's very unintuitive and confusing to flip these around and use 0 for not OK

Reply:
As discussed offline, I agree if we look at the metric as a `exit-code`, this is how it was originally implemented. We switched to view the value more as `true`/`false` or `up`/`down`, where this logic makes more sense.

I'm happy to revisit that once we extend this, but it's not a critical part of the system just now.